src/xenia/kernel/xam/content_manager.cc

```diff
@@ -138,6 +146,53 @@ bool ContentManager::ContentExists(const ContentData& data) {
   return std::filesystem::exists(path);
 }
 
+X_RESULT ContentManager::CreateContentHeaderFile(const std::string file_name,
+                                                 const XCONTENT_DATA* data_raw) {
+  auto title_id = fmt::format("{:8X}", kernel_state_->title_id());
+  auto header_path = root_path_ / title_id / kGameContentHeaderDirName;
+
+  if (!std::filesystem::exists(header_path)) {
+    if (!std::filesystem::create_directories(header_path)) {
+      return X_STATUS_ACCESS_DENIED;
+    }
+  }
+  auto header_filename = file_name + ".header";
+
+  xe::filesystem::CreateFile(header_path / header_filename);
+
+  if (std::filesystem::exists(header_path / header_filename)) {
+    auto file = xe::filesystem::OpenFile(header_path / header_filename, "wb");
+    fwrite(data_raw, 1, sizeof(XCONTENT_DATA), file);
+    fclose(file);
+    return X_STATUS_SUCCESS;
+  }
+  return X_STATUS_NO_SUCH_FILE;
+}
+
+X_RESULT ContentManager::OpenContentHeaderFile(const std::string_view file_name,
+                                           XCONTENT_DATA& data) {
+  auto title_id = fmt::format("{:8X}", kernel_state_->title_id());
+  auto header_file_path =
+      root_path_ / title_id / kGameContentHeaderDirName / file_name;
+
+  if (std::filesystem::exists(header_file_path)) {
+    auto file = xe::filesystem::OpenFile(header_file_path, "rb");
+
+    std::vector<uint8_t> buffer;
+
+    fseek(file, 0, SEEK_END);
```

Comment:
Error handling (just ifs and returns) after every stdio call would be good for safety.